packages/block-editor/src/components/media-replace-flow/index.js

```diff
@@ -112,6 +111,15 @@ const MediaReplaceFlow = ( {
 		isAlternate: true,
 	};
 
+	let name = __( 'Replace' );
+	if ( allowedTypes[ 0 ] === 'image' && allowedTypes.length === 1 ) {
```

Comment:
I have a feeling there's a much nicer way to do this, but as a draft PR this gets the point across.

Reply:
You will need to set a default for allowedTypes to an empty array to avoid errors when nothing is passed. Then use array destructuring:

```js
const [ firstType ] = allowedTypes;

if ( firstType === 'image' ) {
   //...
}
```

You can also use `switch` statement:

```js
let name = __( 'Replace' );
const [ firstType ] = allowedTypes;

switch ( firstType ) {
  case 'image':
	  name = __( 'Change image' );
	  break;
  case 'video':
	  name = __( 'Change video' );
	  break;
  case 'audio':
	  name = __( 'Change audio' );
	  break;
}
```

Reply:
P.S. I don't mean these are "nicer," just alternative ways of doing things 😄

Reply:
Thank you! Those _are_ nicer, much appreciated. I'll adopt one when I get a moment 👌